.github/workflows/sbom.yaml

```diff
@@ -0,0 +1,41 @@
+name: Generate SBOM with Kubernetes BOM
+
+on:
+  release:
+    types:
+      - published
+
+permissions:
+  contents: read
+
+jobs:
+  sbom:
+    runs-on: ubuntu-latest
+
+    permissions:
+      contents: write
+
+    env:
+      OUTPUT: kube-state-metrics-${{ github.ref_name }}.spdx
+      TAG: ${{ github.ref_name }}
+
+    steps:
+      - name: Fetch source code into GITHUB_WORKSPACE
+        uses: actions/checkout@1d96c772d19495a3b5c517cd2bc0cb401ea0529f # v4.1.3
+
+      - name: Install Kubernetes BOM
+        uses: kubernetes-sigs/release-actions/setup-bom@841d76a188a7c121231a863572e27012805715a2 # v0.1.4
+
+      - name: Generate SBOM
+        run: |
+          bom generate \
+            --dirs=. \
+            --image=registry.k8s.io/kube-state-metrics/kube-state-metrics:$TAG \
```

Comment:
Does this assume that the tag is already in the registry and built during release?

Reply:
Yes, it assumes the container image has been tagged already with the same tag used for GitHub Release.

Is this safe to assume?

Reply:
See our RELEASE.md, we usually create a pre-release and then have another PR to get the container images promoted from the staging registry.

Reply:
As far as I could understand, by the time a pre-release is marked as a regular release, the image was already promoted with the proper tag:

> Once the PR to promote the image is merged, mark the pre-release as a regular release.

https://github.com/kubernetes/kube-state-metrics/blob/main/RELEASE.md

Therefore, using `types: [released]` instead of `types: [published]` should resolve this, because the workflow will only trigger when:

> A release was published, or a pre-release was changed to a release.

https://docs.github.com/en/webhooks/webhook-events-and-payloads?actionType=released#release

Should be resolved with 2a23510

What do you think?